### PR TITLE
Feature/strategy serialization

### PR DESCRIPTION
**Pull request by evo-bschiller**
In this PR, we add serialization of strategies and all related components.

### CODE REVIEW

File: bofire/strategies/botorch/qehvi.py [hunk 1]
```diff
@@ -27,17 +27,32 @@
...
+    type: Literal["BoTorchQehviStrategy"] = "BoTorchQehviStrategy"
+
     ref_point: Optional[dict] = None
     ref_point_mask: Optional[np.ndarray] = None
     objective: Optional[MCMultiOutputObjective] = None
 
+    # TODO: unhide fields
+    def json(self, **kwargs):
+        return super().json(
+            **add_exclude(kwargs, "ref_point", "ref_point_mask", "objective")
```
Comment (jduerholt) [inline bofire/strategies/botorch/qehvi.py:47]: the ref point should not be excluded,

File: bofire/models/torch_models.py [hunk 2]
```diff
@@ -250,8 +260,8 @@ def compatibilize(
                     model.model.input_transform = features_filter  # type: ignore
 
                 botorch_models.append(model.model)
-                if isinstance(model.model, botorch.models.SingleTaskGP) is False:
-                    all_gp = False
+            if isinstance(model.model, botorch.models.SingleTaskGP) is False:
```
[the remark this answers is not in the document]
Comment (evo-bschiller) [reply bofire/models/torch_models.py:263]: seems not to be in the most current version, please check and re-comment if the issue still persists

File: bofire/strategies/botorch/qehvi.py (hunk 1 above)
Comment (evo-bschiller) [reply bofire/strategies/botorch/qehvi.py:47]: yes, it is currently hidden to allow for correct serialization.
will be "fixed" later 8see the TODOs).

File: bofire/models/torch_models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jduerholt) [reply bofire/models/torch_models.py:263]: https://github.com/experimental-design/bofire/blob/1ebeb8e87cb96e1035f423c4817a1b3b82cf0435/bofire/models/torch_models.py#L253 This is how it is currently.

**Review by jduerholt — APPROVED** [review]
Looks good to me.

**Pull request merged**